packages/js/src/ui/helpers/createInfiniteScroll.ts

```diff
@@ -48,6 +49,18 @@ export function createInfiniteScroll<T>(fetcher: (page: number) => Promise<{ dat
     });
   });
 
+  const reset = () => {
+    setData([]);
+    setInitialLoading(true);
+    setEnd(false);
+
+    if (offset() !== 0) {
+      setOffset(0);
+    } else {
+      refetch();
+    }
+  };
```

Comment:
The function allows the infinite scroll state to be reset. When the offset is different than 0 it's enough just to update the offset state so it will trigger the fetcher, otherwise, we need to call `refetch` manually which will call the fetcher.